community/faas/docs/testing.md

```diff
@@ -0,0 +1,77 @@
+# tests
+
+This project's consistency and quality control are backed by simple integration
+tests using the [Bats testing framework](https://github.com/sstephenson/bats).
+
+
+## Requirements
+
+To install bats, follow the instructions in the Bats website above, or execute
+the *development make target*:
+
+```
+make development
+```
+
+
+## Setting up the testing environment
+
+Since most companies and persons using or developing against this project will
+be working under a different GCP Organization, some of the base settings needed
+for these tests to run were abstracted, and moved to a *personalized*
+configuration file, which by default is expected to be `~/.faas-tests.conf`,
+though the location of this tests config file can be customized by setting the
+variable `FAAS_CONF`. For example:
+
+```
+export FAAS_CONF=/etc/faas-tests.conf`
+```
+
+This file needs to be set with the site-specific information for each person or
+organization. An example for this file is under `tests/faas-tests.conf.example`
+
+
+## Running tests
+
+Currently, only specific test files can be executed at at time. Work is being
+done to implement a full suite test runner that will execute all (or a subset)
+of the tests available.
+
+Paths are important. Always run the test from the root of the `faas` project:
+
+```
+./tests/integration/network.bats
```

Comment:
I think we should compartmentalize tests. Ex. ./network/tests/integration/network.bats. @ocsig thoughts?

Reply:
For now, I'd suggest is not to compartmentalize them just yet until this project gets a little bit more mature. If we want to reuse testcases/fixtures, the implementation will be quite a bit more complex, pressuring our timelines. Please let us know if this is a must do for this PR.

Reply:
Fine.